src/main/scala/is/hail/expr/Type.scala

```diff
@@ -665,19 +667,38 @@ abstract class TContainer extends Type {
       _elementsOffset(length)
   }
 
+  def elementsOffset(length: Code[Int]): Code[Long] = {
+    val alignment = elementType.alignment
+    assert(alignment > 0, s"invalid alignment: $alignment")
+    assert((alignment & (alignment - 1)) == 0, s"invalid alignment: $alignment") // power of 2
```

Comment:
no messages in asserts, sadly, because Scala will allocate an object for every message each time `elementsOffset` is called.